File: src/main/java/com/tranhuudat/prms/service/SkillService.java
package com.tranhuudat.prms.service;

import com.tranhuudat.prms.dto.BaseResponse;
import com.tranhuudat.prms.dto.skill.SkillDto;
import com.tranhuudat.prms.dto.skill.SkillSearchRequest;

import java.util.UUID;

public interface SkillService {
    BaseResponse create(SkillDto request);

    BaseResponse update(UUID id, SkillDto request);

    BaseResponse delete(UUID id);

    BaseResponse getById(UUID id);

    BaseResponse getPage(SkillSearchRequest request);
}

File: prms-web/src/app/pages/management/skills/services/skill.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from '../../../../core/services/base-service';
import { SkillWritePayload } from '../models/skill.model';
import { SkillSearchRequest } from '../models/skill-search.request';

@Injectable({
  providedIn: 'root',
})
export class SkillService {
  private readonly apiUrl = '/api/v1/skills';

  constructor(private base: BaseService) {}

  getPage(request: SkillSearchRequest) {
    return this.base.post(this.apiUrl + '/page', request);
  }

  getById(id: string) {
    return this.base.get(`${this.apiUrl}/${id}`);
  }

  create(request: SkillWritePayload) {
    return this.base.post(this.apiUrl, request);
  }

  update(id: string, request: SkillWritePayload) {
    return this.base.put(`${this.apiUrl}/${id}`, request);
  }

  delete(id: string) {
    return this.base.delete(`${this.apiUrl}/${id}`);
  }
}

